# HG changeset patch
# User Raphaël Gomès <rgomes@octobus.net>
# Date 1729684194 -7200
#      Wed Oct 23 13:49:54 2024 +0200
# Branch stable
# Node ID d906406658a947ab64b34302df15be21e928ba24
# Parent  e760a36a601336f75016e5c2bbc5a53da2ea7182
patchbomb: don't test ambiguous address

This is a bug in Python's `email` package and shouldn't be relied on.
Python 3.12 has fixed this problem¹ and raises an exception. We keep the
multiple `-t` because this is still relevant for testing.

[1] https://github.com/python/cpython/issues/102988
[2] https://docs.python.org/3/whatsnew/changelog.html

diff --git a/tests/test-patchbomb.t b/tests/test-patchbomb.t
--- a/tests/test-patchbomb.t
+++ b/tests/test-patchbomb.t
@@ -2414,7 +2414,7 @@
   
 
 test multi-address parsing:
-  $ hg email --date '1980-1-1 0:1' -m tmp.mbox -f quux -t 'spam<spam><eggs>' \
+  $ hg email --date '1980-1-1 0:1' -m tmp.mbox -f quux -t 'spam<spam>' \
   >  -t toast -c 'foo,bar@example.com' -c '"A, B <>" <a@example.com>' -s test -r 0 \
   >  --config email.bcc='"Quux, A." <quux>'
   this patch series consists of 1 patches.
@@ -2435,7 +2435,7 @@
   User-Agent: Mercurial-patchbomb/* (glob)
   Date: Tue, 01 Jan 1980 00:01:00 +0000
   From: quux
-  To: =?iso-8859-1?q?spam?= <spam>, eggs, toast
+  To: =?iso-8859-1?q?spam?= <spam>, toast
   Cc: foo, bar@example.com, =?iso-8859-1?q?A=2C_B_=3C=3E?= <a@example.com>
   Bcc: =?iso-8859-1?q?Quux=2C_A=2E?= <quux>
   
